### DISCUSSION TITLE
Bitclout python library

### DISCUSSION THREAD

**Discussion by F041** [deso-protocol/dips#19] [Q&A]
Creation of a library like the one [steem](https://steem.readthedocs.io/en/latest/install.html) has, in a way it wraps and expands functions provided in https://devsclout.hashnode.dev/

**Comment by HPaulson** [comment]
I'm not quite sure this should belong in CIPs. Additionally, what would such a library even be used for? Are you just looking for an API wrapper? This should likely just be built by a community dev & adopted as such, rather than having the core team spend their time making a wrapper for each language

**Reply by F041** [reply]
Yes, I didn't know it had that name but I think yes.

**Reply by F041** [reply]
One could use the library to get automated reports, for example. Like they do in steem.

**Comment by tijno** [comment]
I think building of various language specific wrappers ontop of core/backend are community projects outside of scope for Bitclout CIP.

Similar to me building a new frontend ontop, or @HPaulson building Prism ontop of bitclout. 

Eg there could be so many: php, nim, python, nodejs, .net

For example ItsAditya has started on devsclout with a python one as you mention @F041 

In addition there are tools like cloutcli building ontop of core golang.

**Reply by dharmeshs** [reply]
Would love a Python wrapper on top of core.  I think that and a nodejs one could open up the pool of devs significantly.

**Reply by HPaulson** [reply]
Yeah -- this is outside the scope of CIPs, and should be done by community engineers.
For example, Prism has a full web-based wrapper for identity, all backend API calls, and etc (Will be open-source soon)

Things like this, and other packages, should all be done and adopted by the community -- not really in the scope of the core team besides maybe adding them to docs

**Reply by ItsAditya-xyz** [reply]
@dharmeshs yup, i have been making one python module !

**Comment by maebeam** [comment]
I'm going to move this into the Q&A section because, as others have mentioned, asking to build a tool on top of the existing APIs and implementations isn't exactly a CIP.

I am glad that you are discussing it though because the core devs and community members can provide guidance on implementation. If you have specific implementation ideas / questions let's discuss!
